src/osal/lv_pthread.c

```diff
@@ -64,7 +64,12 @@ lv_result_t lv_thread_delete(lv_thread_t * thread)
 
 lv_result_t lv_mutex_init(lv_mutex_t * mutex)
 {
-    int ret = pthread_mutex_init(mutex, NULL);
+    pthread_mutexattr_t attr;
+
+    pthread_mutexattr_init(&attr);
+    pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
+    int ret = pthread_mutex_init(mutex, &attr);
+
```

Comment:
```suggestion
    pthread_mutexattr_destroy(&attr);

```

> The pthread_mutexattr_init() call results in the allocation of an opaque object. If the object is not destroyed, a memory leak will result.

https://docs.oracle.com/cd/E19683-01/806-6867/sync-83/index.html

Reply:
Great catch! Fixed!